data/db/clickhouse.py

```diff
@@ -0,0 +1,183 @@
+                                                                                   ['relationship']['to']]
+
+                if clickhouse_schema[s]['relationship']['from'] == clickhouse_schema[s]['relationship']['to']:
+                    clickhouse_schema[s]['relationship']['from'] += '_1'
+                    clickhouse_schema[s]['relationship']['to'] += '_2'
+
+                clickhouse_schema[s]['relationship']['from'] += '_id'
+                clickhouse_schema[s]['relationship']['to'] += '_id'
+
+        return clickhouse_schema
+
+    def get_connection():
+        return Clickhouse.__connection
+
+    def get_schema():
+        return Clickhouse.__schema
+
+    def process_json_line(self, json_line, properties):
+        data = json.loads(json_line)
+        processed_data = []
+
+        for key in properties:
+            # biocypher heritage for number cases
+            if key not in data:
+                key += ':long'
+
+            # certain properties might not exist in all records, setting them to None
+            value = data.get(key)
+
+            # clickhouse discourages the use of NULL, using '' instead
+            if value == None or value == []:
+                value = ''
+            processed_data.append(value)
+
+        processed_data.append(data['_key'])
+        if data.get('_from') is not None and data.get('_to') is not None:
+            # removing collection prefix from _from and _to. E.g. genes/ENSG00000148584 => ENSG00000148584
+            processed_data.append(data['_from'].split('/')[-1])
+            processed_data.append(data['_to'].split('/')[-1])
+
+        return processed_data
+
+    def import_jsonl_file(self, original_jsonl_filepath, collection):
+        client = Clickhouse.__connection
+
+        schema = Clickhouse.get_schema().get(collection)
+        if schema is None:
+            raise ValueError('Collection not defined in schema_config.yaml')
+
+        properties = schema.get('properties')
+
+        sql_properties = ','.join(properties) + ',id'
+        if schema.get('relationship'):
+            sql_properties += ',' + schema['relationship'].get['from']
+            sql_properties += ',' + schema['relationship'].get['to']
+
+        print('Loading data...')
+
+        bulk = []
+        for json_line in open(original_jsonl_filepath, 'r'):
+            bulk.append(tuple(self.process_json_line(json_line, properties)))
+
+            if len(bulk) == 10000:
+                try:
+                    client.execute('INSERT INTO ' + collection +
+                                   ' (' + sql_properties + ') VALUES', bulk)
+                    bulk.clear()
+                except:
+                    # every collection has particular edge cases
+                    # this is needed until we have all collections loaded
+                    import pdb
+                    pdb.set_trace()
```

Comment:
for now, I thought about leaving this until we ingest all collections into Clickhouse. Every dataset has its own small edge cases and we will have to adapt to each one of them.

I can also remove this for the PR, please let me know your thoughts.

Reply:
sounds good